AccountMasterKey.js

```diff
@@ -158,24 +159,32 @@ export class AccountMasterKey {
    * @param {Object} options - The options to use.
    * @param {string} options.accountId - The ID of the account associated with
    *   this master key.
-   * @param {Object} options.kmsService - The kmsService to use to perform key
+   * @param {Object} [options.service] - The kmsService to use to perform key
    *   operations.
    * @param {string} options.kmsPlugin - The ID of the KMS plugin to use.
+   * @param {string} [options.secret = null]
+   * - A secret that is used to generate a key.
    *
    * @returns {Promise<AccountMasterKey>} The new AccountMasterKey instance
    *   or `null` if no cached key for `accountId` could be loaded.
    */
-  static async fromCache({accountId, kmsService, kmsPlugin}) {
+  static async fromCache({accountId, service, kmsPlugin, secret = null}) {
+    const kmsService = service || new KmsService();
     if(typeof localStorage === 'undefined') {
       return null;
     }
 
     const seed = await _seedCache.get({accountId});
     if(!seed) {
+      if(secret !== null) {
```

Comment:
I think we need to put this outside of the seed check or we could load the wrong account master key (a stale one that doesn't match the given `secret` from the cache).

Reply:
moved it outside of the seed check.